core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -171,6 +171,7 @@ public static boolean createNewFile(
       final FsPermission permission) throws IOException {
     return getCarbonFile(filePath).createNewFile(filePath, fileType, doAs, permission);
   }
+
```

Comment:
How about append support? please add subtasks to CARBONDATA-1827:
1. support S3 table with dictionary
2. support compaction on S3 table
3. support data update/delete on S3 table
4. support alter table add columns/drop columns on S3 table